compyle/cuda.py

```diff
@@ -35,7 +35,11 @@ def set_context():
 
 class SourceModule(_SourceModule):
     def __getattr__(self, name):
-        return self.get_function(name)
+        def kernel(*args, **kwargs):
+            f = self.get_function(name)
+            return f(*args, **kwargs)
+        kernel.function_name = name
```

Comment:
Can't you set `kernel.__name__` do you need another attribute?

Reply:
The pyopencl.Program uses function_name so I wanted to keep it the same for cuda kernels as well